frontend/src/pages/integrations/bitbucket/create.tsx

```diff
@@ -1,196 +1,310 @@
+      createNotification({
+        type: "success",
+        text: "Successfully created integration"
+      });
+      router.push(`/integrations/${currentWorkspace?.id}`);
     } catch (err) {
+      createNotification({
+        type: "error",
+        text: "Failed to create integration"
+      });
       console.error(err);
     }
   };
 
-  return integrationAuth &&
-    workspace &&
-    selectedSourceEnvironment &&
-    integrationAuthApps &&
-    targetEnvironments ? (
-    <div className="flex h-full w-full items-center justify-center">
-      <Card className="max-w-md rounded-md p-8">
-        <CardTitle className="text-center">BitBucket Integration</CardTitle>
-        <FormControl label="Project Environment" className="mt-4">
-          <Select
-            value={selectedSourceEnvironment}
-            onValueChange={(val) => setSelectedSourceEnvironment(val)}
-            className="w-full border border-mineshaft-500"
-          >
-            {workspace?.environments.map((sourceEnvironment) => (
-              <SelectItem
-                value={sourceEnvironment.slug}
-                key={`source-environment-${sourceEnvironment.slug}`}
+  useEffect(() => {
+    if (!bitbucketRepos || !bitbucketWorkspaces || !currentWorkspace) return;
+
+    reset({
+      targetRepo: bitbucketRepos[0],
+      targetWorkspace: bitbucketWorkspaces[0],
+      sourceEnvironment: currentWorkspace.environments[0],
+      secretPath: "/",
+      scope: ScopeOptions[0]
+    });
+  }, [bitbucketWorkspaces, bitbucketRepos, currentWorkspace]);
+
+  if (isProjectLoading || isBitbucketWorkspacesLoading || isBitbucketReposLoading)
+    return (
+      <div className="flex h-full w-full items-center justify-center p-24">
+        <Spinner />
+      </div>
+    );
+
+  const scope = watch("scope");
+
+  return (
+    <form
+      onSubmit={handleSubmit(onSubmit)}
+      className="flex h-full w-full items-center justify-center"
+    >
+      <Card className="max-w-md rounded-md p-8 pt-4">
+        <CardTitle className=" text-center">
+          <SiBitbucket size="1.2rem" className="mr-2 mb-1 inline-block" />
+          Bitbucket Integration
+        </CardTitle>
+        <Controller
+          control={control}
+          name="sourceEnvironment"
+          render={({ field: { value, onChange }, fieldState: { error } }) => (
+            <FormControl
+              errorText={error?.message}
+              isError={Boolean(error)}
+              label="Project Environment"
+            >
+              <MultiSelect
+                closeMenuOnSelect
+                isMulti={false}
+                getOptionValue={(option) => option.slug}
+                value={value}
+                getOptionLabel={(option) => option.name}
+                onChange={onChange}
+                options={currentWorkspace?.environments}
+                placeholder="Select a project environment"
+                isDisabled={!bitbucketWorkspaces?.length}
+              />
```

Comment:
our combobox is trash die to limitations with our react - this single select works essentially like a combox and has the desired behavior - will rename though. FLA requested filtering for this integration and since all these can have n number of options I thought it'd be best to just use this for all fields